core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -207,21 +207,65 @@ private TrustAnchor getTrustAnchorForPublicKey(final JsonObject keyProps) {
         }
     }
 
+    /**
+     * Gets the list of configured adapters for the tenant.
+     *
+     * @return The list of configured adapters or {@code null} if not set.
+     */
+    @JsonIgnore
+    public List<Adapter> getAdapters() {
+        return adapters;
+    }
+
+    /**
+     * Sets the given list of adapters for the tenant.
+     *
+     * @param adapters The list of adapters or {@code null} if not set.
+     * @return This tenant for command chaining.
+     */
+    @JsonIgnore
+    public TenantObject setAdapters(final List<Adapter> adapters) {
+        this.adapters = adapters;
+        return this;
+    }
+
+    /**
+     * Gets the adapter configuration for the given type.
+     *
+     * @param type The adapter's type.
+     * @return The adapter configuration or {@code null} if not set.
+     */
+    @JsonIgnore
+    public Adapter getAdapter(final String type) {
+
+        Objects.requireNonNull(type);
+
+        return Optional.ofNullable(adapters)
+                .flatMap(ok -> adapters.stream()
+                        .filter(adapter -> type.equals(adapter.getType()))
+                        .findFirst())
+                .orElse(null);
+    }
+
     /**
      * Gets the configuration information for this tenant's
      * configured adapters.
      * 
      * @return An unmodifiable list of configuration properties or
      *         {@code null} if no specific configuration has been
      *         set for any protocol adapter.
+     * @throws IllegalArgumentException if the conversion of the adapter object fails.
+     * @deprecated Use {@link #getAdapters()} instead.
      */
+    @Deprecated
     @JsonProperty(TenantConstants.FIELD_ADAPTERS)
     public List<Map<String, Object>> getAdapterConfigurationsAsMaps() {
-        if (adapterConfigurations == null) {
-            return null;
-        } else {
-            return adapterConfigurations;
-        }
+        return Optional.ofNullable(adapters)
```

Comment:
IMHO the `@JsonProperty` annotation should be moved to either the *getAdapters* method or the *adapters* field ...

Reply:
At first I did add Json annotation to the _setAdapters_ and _getAdapters_ methods and then I reverted back. As I was not sure if it is allowed now or only changes as part of release 2.0.0.

Reply:
Hmm, the only thing that might break is a custom annotation processor that someone has built for this annotation in this class.

@dejanb @ctron any thoughts on this?